### DISCUSSION TITLE
cgal polyhedron demo (mean curvature flow skeletonization)

### DISCUSSION THREAD

**Discussion by zahrakheyrollahi** [CGAL/cgal#7836] [General]
## Issue Details
hello
i am trying to extract skeleton and visualyze it.
i use the code from this site 
(https://doc.cgal.org/latest/Surface_mesh_skeletonization/index.html)
and it works،And I got outputs like picture:
![IMG_20231019_172531_385](https://github.com/CGAL/cgal/assets/147084302/0caab2dc-6d40-41d9-908b-1299299e248b)
But I want to output like the following figures
![main_image_suggestion](https://github.com/CGAL/cgal/assets/147084302/74d20076-b93c-454d-bcfd-e7b7f1015717)
I installed polyhedron demo, but I don't know how to give it input and apply the skeleton method
![IMG_20231022_191444_530](https://github.com/CGAL/cgal/assets/147084302/bad7ed23-1a4e-4133-9c86-d934d41fee8e)

## Environment

* Operating system (Windows 64 bits
wsl ubuntu 22.04):
* CGAL version:5.6
* Boost version:1_83_0

**Comment by afabri** [comment]
You have to load a file first.  And then you will find skeletonization in the Operations drop down menu.

**Comment by zahrakheyrollahi** [comment]
Hello, thank you for your reply. I was able to get the mcf (mean curvature skeletons) output.
![IMG_20231022_220752_551](https://github.com/CGAL/cgal/assets/147084302/ad9d893a-fa8b-4ac2-8d16-25a877f95cf9)
 My question is how can I get medial axis outputs like the figure in part b?
![Screenshot_20231024_222704_Adobe Acrobat](https://github.com/CGAL/cgal/assets/147084302/f35b9687-b521-469f-b48c-496ec77f1482)

**Comment by sloriot** [comment]
This is only the intermediate state of the mesh that is contracted along the mean curvature flow. You can see it evolving by calling "Run one iteration".

**Comment by zahrakheyrollahi** [comment]
Is there a separate algorithm or code (without skeleton) in cgal to implement the medial axis in three dimensions?

**Comment by sloriot** [comment]
We only have this package that is able to compute skeleton in 3D.
